lib/charms/traefik_k8s/v2/ingress.py

```diff
@@ -303,40 +315,61 @@ def wipe_ingress_data(self, relation: Relation):
             return
         del relation.data[self.app]["ingress"]
 
-    def _get_requirer_data(self, relation: Relation) -> RequirerData:  # type: ignore
+    def _get_requirer_followers_data(self, relation: Relation) -> List[RequirerUnitData]:
         """Fetch and validate the requirer's app databag.
 
-        For convenience, we convert 'port' to integer.
+        We cast port to int for convenience.
         """
-        if not relation.app or not relation.app.name:  # type: ignore
-            # Handle edge case where remote app name can be missing, e.g.,
-            # relation_broken events.
-            # FIXME https://github.com/canonical/traefik-k8s-operator/issues/34
-            return {}
+        out: List[RequirerUnitData] = []
+
+        unit: Unit
+        for unit in relation.units:
+            databag = relation.data[unit]
+            remote_unit_data: Dict[str, Optional[Union[int, str]]] = {}
+            for key in ("host", "port"):
+                remote_unit_data[key] = databag.get(key)
+            remote_unit_data["port"] = (
+                int(remote_unit_data["port"]) if remote_unit_data["port"] else None
+            )
+            _validate_data(remote_unit_data, INGRESS_REQUIRES_UNIT_SCHEMA)
+            out.append(typing.cast(RequirerUnitData, remote_unit_data))
+        return out
+
+    def _get_requirer_leader_data(self, relation: Relation) -> RequirerAppData:
+        """Fetch and validate the requirer's app databag."""
+        app: Optional[Application] = relation.app
+        if app is None:
+            raise NotReadyError(relation)
```

Comment:
it's too deep in the library code to surface to the charm in any other meaningful manner I feel, and I'd rather people code defensively and do `if ingress.is_ready: data=ingress.get_data` rather than write their code to handle a possible None being returned from `get_data`.

But if we don't like it we can revert